File: Sorting/Ruby/selection_sort.rb
def selection_sort(arr) 
    (arr.length - 1).times do |i|
        min_idx = i
        (i...arr.length).times do |j|
            if arr[min_idx] > arr[j] 
                min_idx = j 
            end
        end
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    end
end

arr = [64, 25, 12, 22, 11]
p selection_sort(arr)

File: Sorting/JavaScript/selectionSort.js
const selectionSort = (arr) => {
    let leftIndex = 0;
    let minIndex = 0;
    let startingIndex = 1;

    while (leftIndex !== arr.length - 1) {
        for(let i = leftIndex; i < arr.length; i++) {
            if(arr[i] < arr[minIndex]) {
                minIndex = i;
            }
        }
        temp1 = arr[leftIndex];
        arr[leftIndex] = arr[minIndex];
        arr[minIndex] = temp1;
        leftIndex += 1;
    };

    return arr;
};

arr = [64, 25, 12, 22, 11];
console.log(selectionSort(arr));